scale/job/configuration/configuration/job_configuration.py

```diff
@@ -137,332 +124,143 @@ def __init__(self, configuration=None):
-
-        if 'pre_task' not in self._configuration:
-            self._configuration['pre_task'] = {}
-        if 'docker_params' not in self._configuration['pre_task']:
-            self._configuration['pre_task']['docker_params'] = []
-        if 'workspaces' not in self._configuration['pre_task']:
-            self._configuration['pre_task']['workspaces'] = []
-
-        if 'docker_params' not in self._configuration['job_task']:
-            self._configuration['job_task']['docker_params'] = []
-        if 'workspaces' not in self._configuration['job_task']:
-            self._configuration['job_task']['workspaces'] = []
-
-        if 'post_task' not in self._configuration:
-            self._configuration['post_task'] = {}
-        if 'docker_params' not in self._configuration['post_task']:
-            self._configuration['post_task']['docker_params'] = []
-        if 'workspaces' not in self._configuration['post_task']:
-            self._configuration['post_task']['workspaces'] = []
-
-    def _validate_workspace_names(self):
-        """Ensures that no tasks have duplicate workspace names
+    def _validate_setting_names(self):
+        """Ensures that no tasks have duplicate setting names
 
         :raises :class:`job.configuration.configuration.exceptions.InvalidJobConfiguration`: If there is a duplicate
             workspace name
         """
 
-        for workspace_dict in self._configuration['pre_task']['workspaces']:
-            name = workspace_dict['name']
+        for setting_dict in self._configuration['pre_task']['settings']:
+            name = setting_dict['name']
             if name in self._pre_task_workspace_names:
-                raise InvalidJobConfiguration('Duplicate workspace %s in pre task' % name)
+                raise InvalidJobConfiguration('Duplicate setting %s in pre task' % name)
             self._pre_task_workspace_names.add(name)
-        for workspace_dict in self._configuration['job_task']['workspaces']:
-            name = workspace_dict['name']
+        for setting_dict in self._configuration['job_task']['settings']:
+            name = setting_dict['name']
             if name in self._job_task_workspace_names:
-                raise InvalidJobConfiguration('Duplicate workspace %s in job task' % name)
+                raise InvalidJobConfiguration('Duplicate setting %s in job task' % name)
             self._job_task_workspace_names.add(name)
-        for workspace_dict in self._configuration['post_task']['workspaces']:
-            name = workspace_dict['name']
+        for setting_dict in self._configuration['post_task']['settings']:
+            name = setting_dict['name']
             if name in self._post_task_workspace_names:
-                raise InvalidJobConfiguration('Duplicate workspace %s in post task' % name)
+                raise InvalidJobConfiguration('Duplicate setting %s in post task' % name)
             self._post_task_workspace_names.add(name)
```

Comment:
Need to check each task to see if any settings are duplicated. Also please fix the docstring (mentions workspace names)